### ISSUE TITLE
Metrics from playwright to honeycomb

### ISSUE THREAD

**Issue by vikramjb** [artilleryio/artillery#2529]
Version info: 
```
2.0.6
```

Running this command: 
```
npx artillery run .\loaddefinition\test-load.yml
```

I expected to see this happen:

*Expecting to see the metrics getting uploaded to honeycomb*

Instead, this happened:

*I can see the tests running but no metrics are being uploaded to honeycomb. I tried commenting the apiKey and writeKey and sure enough validation kicked in. However when running with the api keys in the yaml file, I do not see the metrics being pushed to Honeycomb*

Files being used: 

**test-load.yml**
```
config:
  target: https://google.com
  phases:
    - duration: 30
      arrivalRate: 2
  processor: "../tests/testflow.js"
  engines:
    playwright:
      useSeparateBrowserPerVU: false
      aggregateByName: true
      defaultTimeout: 120
      defaultNavigationTimeout: 120
      extendedMetrics: true
      showAllPageMetrics: false
      launchOptions:
        headless: true
      contextOptions:
         viewport: null
  plugins:
    publish-metrics:
      - type: honeycomb
        apiKey: API_KEY
        writeKey: API_KEY
        dataset: "art-playwright-test1"
        sampleRate: 1
scenarios:
  - engine: playwright
    name: "TestCloudWaitListSignup"
    flowFunction: "cloudWaitlistSignupFlow"
```

**testflow.js**
```

async function cloudWaitlistSignupFlow(page, userContext, events, test) {
    await test.step('Go to Google', async()=>{
      const requestPromise = page.waitForRequest('https://google.com/');
      await page.goto('https://google.com/');
      const req = await requestPromise;
    })
  }
  
  module.exports = {
    cloudWaitlistSignupFlow
  };
  
```

When I run the same honeycomb config for a http engine, it works fine and I can see the report on Honeycomb without any issues.

**Comment by InesNi** [comment]
Hey @vikramjb 👋🏻  Thanks for the thorough report! I'll take a look at this as soon as I'm able to with your test script.

In the meantime, if you have a chance, would you be able to rerun the test with the environment variable `DEBUG=honeycomb,open-telemetry` ,  and letting me know if you see any errors in that debug (or sharing the debug logs)? It could help point to the cause of the issue! Thanks.

**Comment by InesNi** [comment]
Hi @vikramjb 👋🏻 

There was indeed a bug on our side with Playwright and certain `publish-metrics` reporters. I have a PR open with a fix for this. You will be able to try it against a canary version which I'll post here once fixed.

Thanks again!

**Comment by vikramjb** [comment]
Hey @InesNi 👋

Thank you for the quick fix. Eager to try the canary release 👍

**Comment by vikramjb** [comment]
@InesNi  👋  Thank you for the fix. I saw the merge onto main. When is the next release expected ? Alternatively is there documentation on how to run artillery from source? I have asked this question on the discussion as well, apologies if this is not the right forum for it.

**Comment by InesNi** [comment]
Hi @vikramjb,

The fix is now available in Artillery [v2.0.7](https://github.com/artilleryio/artillery/releases/tag/artillery-2.0.7),
Thank you once again for reporting the issue 🙇🏻‍♀️

**Comment by vikramjb** [comment]
@InesNi 👋 Got it. Last night I ran the canary build and tested it. It ran fine and I got what I want. Thank you for fixing it so quickly 👌
